### PR TITLE
various redux thrash fixes

### PR DESCRIPTION
**Pull request by chrisnojima**
Fixes some render thrashing issues when receiving a lot of chat messages
- [x] input area rendered a lot all the time due to `suggestAllChannels` `suggestChannels`. Also some helpers in Constants were only used by this connector and had caching issue
- [x] `syncProps` wasn't throttled at all so if you got hammered with messages your widget would try and update as fast as possible really slowing you down. Instead we only update the remotes every second max
- [x] reduce some unnecessary sends to menubar remote
- [x] treat chats `badgeMap` for big teams badge count as `1` if there's any badge. This doesn't affect the nav count but when we bold a channel we only need to know if its 0/1 so additional typing won't thrash the inbox
- [x] tab-bar thrashed on unused `_fullnames` param
- [x] pulled Tabs out into their own component so only the changed tabs need to render instead of all of them
- [x] unneeded sort of keys in `mapEqual` helper

### CODE REVIEW

File: shared/reducers/chat2.tsx [hunk 1]
```diff
@@ -619,12 +619,24 @@ const reducer = Container.makeReducer<Actions, Types.State>(initialState, {
     const unreadMap = new Map<Types.ConversationIDKey, number>()
     conversations.forEach(({convID, badgeCounts, unreadMessages}) => {
       const key = Types.conversationIDToKey(convID)
-      const count = badgeCounts[badgeKey] || 0
+      // if its big we treat the count as 1 or 0 so additional chats don't redraw us
```
Comment (buoyad) [inline shared/reducers/chat2.tsx]: The mobile convo header uses `badgeMap` to show a badge by the back button, with this change it won't show more messages past the first. Maybe we could just do this memoized in the inbox connector?

Comment (chrisnojima) [reply shared/reducers/chat2.tsx]: the header should just show whats in the nav and not compute it again

Comment (buoyad) [reply shared/reducers/chat2.tsx]: It excludes badges for the current chat. otherwise I think the badge flickers in as you're looking at the thread

**Comment by mmaxim** [comment]
Did something change here recently? I feel like the desktop app is particularly slow today.

**Pull request merged**